include/seqan3/core/detail/all_view.hpp

```diff
@@ -0,0 +1,167 @@
+// -----------------------------------------------------------------------------------------------------
```

Comment:
This is from the STL.

`std::views::all` can also bind rvalues via `views::owning_range`.
Since `views::zip` and `views::join_with` use the new `views::all_t` to properly handle rvalues, I implemented the new behaviour.

A follow-up would be to get rid of `seqan3::views::persist` and use either `seqan3::detail::owning_view` or `seqan3::detail::all{_t}`.

Reply:
https://github.com/seqan/product_backlog/issues/416